lib/ansible/runner.rb

```diff
@@ -1,6 +1,12 @@
 module Ansible
   class Runner
     class << self
+      def available?
+        return @available if defined?(@available)
+
+        @available = system("which ansible-runner &>/dev/null")
```

Comment:
I don't think that `&>/dev/null` is right...

```
>> system("which abcd &>/dev/null")
=> true
>> system("which abcd >/dev/null")
=> false
```

If you want to send stdout to `/dev/null` that's `system("which ansible-runner >/dev/null")`
If you want to send both stdout and stderr to `/dev/null` that's `system("which abcd >/dev/null 2>&1")`

Reply:
`&> ` means "redirect both stdout and stderr to" - it's effectively equivalent to your second one without the double redirection and without being so wordy. See also https://www.gnu.org/software/bash/manual/html_node/Redirections.html#Redirecting-Standard-Output-and-Standard-Error

Reply:
Right that is a bash concept and isn't portable
```
adam@desktop:~$ sh
$ which abcd &>/dev/null
$ echo $?
0
[1] + Done(1)                    which abcd
```

shell thinks that is `&` and starts it in the background and returns 0

Reply:
```
[1] pry(main)> system("which abcd")
=> false
[2] pry(main)> system("which abcd &>/dev/null")
=> false
[3] pry(main)> system("which abcd >/dev/null")
=> false
[4] pry(main)> system("which abcd >/dev/null 2>&1")
=> false
```

Reply:
I still think it is better to use the portable method since there are platforms where the result is the opposite of what we want

Reply:
Ok yes, I agree - I'll update

I think I've done this in other places, so this might require changes elsewhere but those are separate PRs

Reply:
Fixed